### DISCUSSION TITLE
Wtorki z security  - Azure Front Door i Application Gateway

### DISCUSSION THREAD

**Discussion by mgrabarz** [AzureCommunityPL/group#87] [Community Content]
Cześć! W imieniu swoim, @k3aczyk i Microsoft chciałbym Was zaprosić na kolejny webinar z serii "Wtorki z Security". 

Tematem odcinka będą dwie popularne usługi **Azure Front Door** i **Application Gateway** oraz ich zastosowanie w kontekście bezpieczeństwa i użycia w celu zabezpieczenia aplikacji internetowych.

![linkedin Application Gateway i Azure Front Door_](https://user-images.githubusercontent.com/20926456/94533948-206b5680-0240-11eb-89ed-c9e40805d8a6.JPG)

### [Link do rejestracji](https://info.microsoft.com/CE-SCRTY-WBNR-FY21-10Oct-06-ApplicationGatewayandAzureFrontDoor-SRDEM42108_LP01Registration-ForminBody.html).

**Comment by jberezanski** [comment]
@mgrabarz fajna i kształcąca prezentacja, dzięki!

Niestety, mimo że na koniec zaprosiłeś do zadawania pytań, to od razu pojawiło mi się na cały ekran "Prezentacja została zakończona." i nie miałem okazji. Dlatego może zadam je tutaj:
1) Czy we Front Door można podać certyfikat własnego CA, jeśli mamy swoje wewnętrzne PKI, z którego wystawiamy certyfikaty dla usług backendowych?
2) Czy ruch między naszym Front Door a backendem jest odizolowany od innych tenantów Azure? Innymi słowy, czy w hipotetycznej architekturze Front Door -> Application Gateway -> VM sensowne jest robienie SSL termination na FD, czy dopiero na AppGw?
3) Czy zakres funkcjonalności WAF pokrywa się ściśle między Front Door i Application Gateway i w przypadku użycia obu usług wystarczy włączyć WAF tylko na jednej z nich, czy może są jakieś powody, dla których warto by na obu?

**Reply by mgrabarz** [reply]
Hej Kuba, 

Taka ta platforma jest, ze wywala wszystkim jak się nagranie skończy. Próbowaliśmy wgrać takie z 15 minutowym ekranem pożegnalnym ale było już "za późno".

1. Z tego co pamiętam, to nie można. Lista dozwolonych znajduje się [tutaj](https://docs.microsoft.com/en-us/azure/frontdoor/front-door-troubleshoot-allowed-ca), lub możesz poprosić AFD by sam po API wygenerował DigiCert'a.
2. Od innych tenantów nie jest, ale też inni nie mogą go podsłuchać (leci po wewnętrznych kabelkach). W regulowanych rynkach raczej bym nie włączał SSL termination na AFD. Z tym pytaniem wiąże się też zagadnienie whitelistowania ruchu z AFD u ciebie - pisał o tym np @smereczynski tu: https://lnx.azurewebsites.net/azure-app-service-on-linux-php-with-azure-front-door-access-control-configuration/
3. Niestety nie :/ AFD z tego co mi wiadomo selektywnie broni przed adresami IP ze znanych i aktualizowanych źródeł ataku i zwykły kowalski się przebije. W AppGw masz szerszą ochronę.

**Reply by jberezanski** [reply]
Dzięki za odpowiedzi!

1. Uściślając - nie chodzi mi o cert, którym będzie się przedstawiał AFD na zewnątrz, tylko o certy, które mam skonfigurowane na backendzie. Chodzi mi o to, czy mogę nauczyć AFD ufać mojemu CA, z którego wystawiłem sobie certy zainstalowane na moich VMkach czy innych serwisach. W AppGw [mogę](https://docs.microsoft.com/en-us/azure/application-gateway/self-signed-certificates#upload-the-root-certificate-to-application-gateways-http-settings) tak zrobić.

Aha, zajrzałem do [dokumentacji](https://docs.microsoft.com/en-us/azure/frontdoor/front-door-faq#why-is-https-traffic-to-my-backend-failing) i wychodzi, że w AFD _nie_ można - akceptuje backendowe certy tylko wystawione przez CA z tej samej listy, z której należy pozyskiwać certy dla samego AFD. Szkoda.

**Comment by kmadof** [comment]
@mgrabarz Jest może dostępne nagranie?

**Reply by mgrabarz** [reply]
Jeżeli się rejestrowałeś, to możesz obejrzeć on-demand w dowolnym momencie. 

Jeśli nie, to spróbuj tutaj:  https://info.microsoft.com/CE-SCRTY-CATALOG-FY21-07Jul-29-WtorkizSecurity-SRDEM32279_CatalogDisplayPage.html
